### PR TITLE
fix(ts): add type definition for functional babel.presets

### PR DESCRIPTION
**Pull request by clarkdo**
## Types of changes

- [x] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] I have added tests to cover my changes (if not applicable, please state why)
- [ ] All new and existing tests are passing.

### CODE REVIEW

File: packages/config/types/build.d.ts [hunk 1]
```diff
@@ -18,9 +18,17 @@ import { TerserPluginOptions } from 'terser-webpack-plugin'
 
 type NuxtConfigurationLoaders = any // TBD
 
+interface NuxtBabelPresetEnv {
+  isServer: boolean;
+}
+
+interface NuxtBabelOptions extends Pick<TransformOptions, Exclude<keyof TransformOptions, 'presets'>> {
+  presets?: ((env: NuxtBabelPresetEnv, presets: [string, object | null]) => PluginItem[] | void) | PluginItem[] | null
```
Comment (kevinmarrec) [inline packages/config/types/build.d.ts]: Is there really a case that we accept functional `presets` returning `void` ? Should it not be always an array ?

Assuming https://github.com/nuxt/nuxt.js/blob/dev/packages/webpack/src/config/base.js#L81 & https://github.com/nuxt/nuxt.js/blob/dev/packages/webpack/src/config/base.js#L89, 
`presets` function parameter should be named `defaultPreset` instead and seems to fit `PluginItem` type.

```suggestion
  presets?: ((env: NuxtBabelPresetEnv, defaultPreset: PluginItem) => PluginItem[]) | PluginItem[] | null
```

**Review by kevinmarrec — DISMISSED** [review]

File: packages/config/types/build.d.ts (hunk 1 above)
Comment (clarkdo) [reply packages/config/types/build.d.ts]: `presets` can modify the second argument and return nothing.

`presets: [string, object | null]` is fixed strcture, so it shouldn't 100% follow the structure of PluginItem

Comment (kevinmarrec) [reply packages/config/types/build.d.ts]: Alright all good !

Comment (clarkdo) [reply packages/config/types/build.d.ts]: Arg name is changed, thanks

File: packages/config/types/build.d.ts [hunk 2]
```diff
@@ -18,9 +18,17 @@ import { TerserPluginOptions } from 'terser-webpack-plugin'
 
 type NuxtConfigurationLoaders = any // TBD
 
+interface NuxtBabelPresetEnv {
+  isServer: boolean;
```
Comment (kevinmarrec) [inline packages/config/types/build.d.ts]:
```suggestion
  isServer: boolean
```

**Review by kevinmarrec — DISMISSED** [review]

**Review by kevinmarrec — APPROVED** [review]

**Pull request merged**